app/views/users/_additional_authentication.html.erb

```diff
@@ -1,16 +1,17 @@
-<% if (SiteConfig.auth_allowed?("github") || SiteConfig.auth_allowed?("twitter")) && (!@user.identities.exists?(provider: "github") || !@user.identities.exists?(provider: "twitter")) %>
-  <div class="crayons-card mb-6 p-6">
-    <div class="flex">
-      <% if SiteConfig.auth_allowed?("github") && !@user.identities.exists?(provider: "github") %>
-        <a href="/users/auth/github" class="crayons-btn crayons-btn--outlined mr-2" data-no-instant>
-          <%= inline_svg_tag("github.svg", aria: true, class: "crayons-icon", title: "GitHub") %>Connect GitHub Account
-        </a>
-      <% end %>
-      <% if SiteConfig.auth_allowed?("twitter") && !@user.identities.exists?(provider: "twitter") %>
-        <a href="/users/auth/twitter?callback_url=<%= app_url("/users/auth/twitter/callback") %>" class="crayons-btn crayons-btn--outlined" data-no-instant>
-          <%= inline_svg_tag("twitter.svg", aria: true, class: "crayons-icon", title: "GitHub") %>Connect Twitter Account
-        </a>
-      <% end %>
-    </div>
-  </div>
+<% Authentication::Providers.enabled.each do |provider_name| %>
+  <% unless @user.identities.exists?(provider: provider_name) %>
+    <% provider = Authentication::Providers.get!(provider_name) %>
+    <a
+      href="<%= Authentication::Providers.authentication_path(provider_name) %>"
+      class="crayons-btn crayons-btn--outlined mr-2"
+      data-no-instant>
+      <%= inline_svg_tag(
+            "#{provider_name}.svg",
```

Comment:
Do we have any checks to ensure that an svg exists for a given provider?

Reply:
Not, not at present :-( 

Also, `inline_svg_tag` won't break:

```ruby
[11] pry(main)> inline_svg_tag("foobar.svg")
=> "<svg><!-- SVG file not found: 'foobar.svg' --></svg>"
```

maybe I can add a spec to check that "SVG file not found" isn't in